app/components/TrackDetailsComponent/index.js

```diff
@@ -0,0 +1,140 @@
+import React, { memo } from 'react';
+import PropTypes from 'prop-types';
+import { Carousel } from 'antd';
+import { isEmpty } from 'lodash';
+import styled from 'styled-components';
+import If from '@components/If';
+import T from '@components/T';
+
+const Details = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    background-color: #231f20;
+    color: white;
+    margin-left: 7rem;
+  }
+`;
+
+const TrackTitle = styled.div`
+  && {
+    text-align: center;
+    margin-top: 3rem;
+    color: white;
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+    align-items: center;
+  }
+`;
+
+const ArtistName = styled.div`
+  && {
+    text-align: center;
+    color: red;
+  }
+`;
+
+const StyledCarousel = styled(Carousel)`
+  && {
+    width: 20rem;
+    height: 25rem;
+    margin-left: 20%;
+  }
+`;
+
+const TrackImage = styled.img`
+  && {
+    border-radius: 0.5rem;
+    height: 20rem;
+    width: 20rem;
+  }
+`;
+
+const AudioPlayer = styled.audio`
+  && {
+    width: 20rem;
+    height: 6rem;
+    color: black;
+    margin: 1rem;
+  }
+`;
+
+const TrackDetailsCard = styled.div`
+  && {
+    display: flex;
+    flex-direction: row;
+  }
+`;
+
+export function TrackDetailsComponent({
+  trackName,
+  previewUrl,
+  artistName,
+  artworkUrl100,
+  collectionName,
+  primaryGenreName,
+  trackViewUrl
+}) {
+  return (
+    <TrackDetailsCard data-testid="track-details-component">
+      <If condition={!isEmpty(artworkUrl100)} otherwise={<T data-testid="image_unavailable" id="image_unavailable" />}>
+        <StyledCarousel autoplay dotPosition="bottom">
+          <div>
```

Comment:
can we avoid this extra dev and the nesting?

Reply:
or. if its necessary can we create another small component that does this instead?